src/lambda/local/configureLocalLambda.ts

```diff
@@ -131,31 +153,77 @@ function getTabSize(editor?: vscode.TextEditor): number {
 
 async function loadSymbols(
     uri: vscode.Uri,
-    maxAttempts = 3,
-    retryDelayMillis = 500
+    context: ConfigureLocalLambdaContext,
+    maxAttempts = 10,
+    retryDelayMillis = 200
 ): Promise<vscode.DocumentSymbol[] | undefined> {
-    if (maxAttempts <= 0) {
-        return undefined
-    }
 
-    const symbols = await vscode.commands.executeCommand<vscode.DocumentSymbol[]>(
+    const symbols = await context.executeCommand<vscode.DocumentSymbol[]>(
         'vscode.executeDocumentSymbolProvider',
         uri
     )
+    // checking if symbols exists as this can fail if the VS Code JSON symbol provider is not yet initialized
     if (symbols) {
+        // file has symbols if JSON with at least one valid top-level key/value pair
         return symbols
     }
 
+    if (maxAttempts <= 0) {
+        return undefined
+    }
+
+    // waiting before retry to wait for JSON parser
     await sleep(retryDelayMillis)
 
-    return await loadSymbols(uri, maxAttempts - 1, retryDelayMillis)
+    return await loadSymbols(uri, context, maxAttempts - 1, retryDelayMillis)
 }
 
-async function prepareConfig(editor: vscode.TextEditor, handler: string): Promise<boolean> {
-    const symbols: vscode.DocumentSymbol[] | undefined = await loadSymbols(editor.document.uri)
+async function prepareConfig(editor: vscode.TextEditor, handler: string,
+                             context: ConfigureLocalLambdaContext): Promise<boolean> {
+    let symbols: vscode.DocumentSymbol[] | undefined = await loadSymbols(editor.document.uri, context)
+    let shouldOverwrite: boolean = false
+    let shouldLoop: boolean = !symbols
+
+    while (!symbols && shouldLoop) {
```

Comment:
Minor: This function is getting pretty long. Suggest to break out some of the functionality into helper methods as you see fit.